src/mono/mono/component/debugger-engine.c

```diff
@@ -364,29 +364,42 @@ typedef struct {
 	GPtrArray *methods;
 	GPtrArray *method_domains;
 	GPtrArray *method_seq_points;
+
+	MonoDomain *domain;
 } CollectDomainData;
 
+static void
+collect_domain_bp_inner (gpointer key, gpointer value, gpointer user_data)
+{
+	MonoMethod *m = (MonoMethod *)key;
+	MonoSeqPointInfo *seq_points = (MonoSeqPointInfo *)value;
+	CollectDomainData *ud = (CollectDomainData*)user_data;
+
+	if (!bp_matches_method (ud->bp, m))
+		return;
+
+	/* Save the info locally to simplify the code inside the domain lock */
+	g_ptr_array_add (ud->methods, m);
+	g_ptr_array_add (ud->method_domains, ud->domain);
+	g_ptr_array_add (ud->method_seq_points, seq_points);
+}
+
+// HACK: Address some sort of linker problem on arm64
+void
+mono_jit_memory_manager_foreach_seq_point (dn_simdhash_ght_t *seq_points, dn_simdhash_ght_foreach_func func, gpointer user_data);
+
 static void
 collect_domain_bp (gpointer key, gpointer value, gpointer user_data)
 {
-	GHashTableIter iter;
-	MonoSeqPointInfo *seq_points;
-	MonoDomain *domain = (MonoDomain*)key;
 	CollectDomainData *ud = (CollectDomainData*)user_data;
-	MonoMethod *m;
+	ud->domain = (MonoDomain*)key;
```

Comment:
We don't support multiple domains anymore. there is only the root domain.  Let's just rearchitect the caller of `collect_domain_bp` to avoid this hash table entirely. /cc @thaystg

Reply:
I can clean that up, thanks.